### ISSUE TITLE
In a local environment, step 4 works but step 5 never populates

### ISSUE THREAD

**Issue by jbuttery** [diafygi/gethttpsforfree#45]
This is for a local install (created by "git clone", not "Save As...") accessed via the file:// protocol in Mozilla Firefox 42.

I can go through the entire process, through Step 4 ("I'm now serving the files") but nothing ever happens for Step 5; wish I could give more info but there's no error or anything.  I did pop open RequestPolicy's log window and I don't see any outgoing requests, so I'm assuming that's related (I have RP's filtering temporarily disabled, as well as NoScript's).  When I click the "I'm now serving the files" button, I get the "Domain verified!" message, but that's the last thing that happens.  It's a two-hostname SAN cert ("domain.com" and "www.domain.com") as per the example OpenSSL command.

When I execute Step 4, I get the following in Firefox' JavaScript console:

ReferenceError: reference to undefined property DOMAINS[domain].confirmed (index.js:802:1)

I had a look at the code there, but everything looks reasonable to me.  I'll keep poking at it, but if you have any ideas on things I could try for troubleshooting, I'm all ears.  :)  Oh, also, I suppose it's worth noting for confirmation's sake that every time I try a new iteration of testing, I do a cache-bypassing Shift-Reload of the page.

**Comment by jbuttery** [comment]
I just added an alert() for debugging into the for loop there:

```
for(var domain in DOMAINS){
    alert(domain);
    if(DOMAINS[domain]['confirmed'] !== true){
        all_confirmed = false;
    }
}
```

...and both hostnames show up in the alert() dialogs as you'd expect.  So, I guess it's the "confirmed" value that's not getting set?

**Comment by diafygi** [comment]
I've added a debug mode. Can you please update your copy of the page and add `?debug` to the url (e.g. https://gethttpsforfree.com/?debug)? After that, please let me know what printed in your console, when you get the error.

**Comment by amn41** [comment]
I'm experiencing the same issue, having just loaded the page from https://gethttpsforfree.com/

Final step seems to be messing up, as console shows a 400 response from lets encrypt https://acme-v01.api.letsencrypt.org/acme/challenge

**Comment by diafygi** [comment]
@amn41 ok, what's the responseText of the 400 response?

**Comment by amn41** [comment]
{"type":"urn:acme:error:malformed","detail":"Unable to read/verify body :: JWS has invalid anti-replay nonce","status":400}

**Comment by diafygi** [comment]
@amn41 can you please clear your cache and reload? the anti-reply nonce issue usually occurs when your browser caches responses

**Comment by jbuttery** [comment]
Well, it's working now, and I think I might be going crazy.  I could swear before, in Step 4, there was just the one section to verify ownership.  I figured it was OK since one SAN name is a subdomain of the other, but it still seemed a bit odd.  Today, though, there are definitely two sections (one per SAN name) and as soon as I did the verify on both of them, Step 5 showed up with the cert instantly.

Did this actually change, or did I somehow manage to miss that there was a verify section per hostname when I was doing this before?
